packages/synapse-react-client/src/components/DatasetJsonLdScript.tsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react'
+import { WithContext, Dataset } from 'schema-dts'
+import useJsonLdScriptElement from '../utils/hooks/useJsonLdScriptElement'
+import MarkdownIt from 'markdown-it'
+import dayjs from 'dayjs'
+import { useGetEntityBundle } from '../synapse-queries'
+import { useWikiPage } from '../synapse-queries/wiki/useWikiPage'
+
+const md = MarkdownIt({ html: true })
+
+interface DatasetJsonLdScriptOption1Props {
+  entityId?: string
+  version?: number
+  searchParams?: never
+}
+interface DatasetJsonLdScriptOption2Props {
+  entityId?: never
+  version?: never
+  searchParams?: { [key: string]: string } // can be injected if this component is embedded in a Dataset Details page
+}
+export type DatasetJsonLdScriptProps =
+  | DatasetJsonLdScriptOption1Props
+  | DatasetJsonLdScriptOption2Props
+
+const getPlainText = (html: string): string => {
+  const divContainer = document.createElement('div')
+  divContainer.innerHTML = html
+  return divContainer.textContent || divContainer.innerText || ''
+}
+
+/**
+ * This component will add a Dataset json ld script tag to the page when rendered.
+ */
+export const DatasetJsonLdScript: React.FunctionComponent<
+  DatasetJsonLdScriptProps
+> = ({
+  entityId: entityIdFromProps,
+  version: versionFromProps,
+  searchParams,
+}) => {
+  let entityId: string
+  let version: number | undefined
+  if (entityIdFromProps) {
+    entityId = entityIdFromProps
+    version = versionFromProps
+  } else if (searchParams) {
+    entityId = searchParams['id']
+    if (searchParams['version']) {
+      version = parseInt(searchParams['version'])
+    }
+  }
+  const { data: entityBundle } = useGetEntityBundle(entityId!, version)
+  const { data: wikiPage } = useWikiPage(entityId!)
+  const html = md.renderInline(wikiPage?.markdown ?? '')
```

Comment:
I could see us re-using and gradually improving a utility to get plain text from Synapse wiki markdown, so you may consider extracting some of this logic into its own file